### PR TITLE
[WIP] Migrates OL auth to use IA creds only

### PR DESCRIPTION
**Pull request by mekarpeles**
@bfalling; further testing required

### CODE REVIEW

File: openlibrary/accounts/model.py [hunk 1]
```diff
@@ -652,15 +654,16 @@ def audit_accounts(email, password, test=False):
...
-        # exists. However, xauthn API doesn't yet support `info` by
-        # itemname. Fortunately, we have all the info we need at this
-        # stage for the client to be satisfied.
         if ol_account.itemname:
-            audit['has_ia'] = ol_account.itemname  # XXX should be email
+            audit['has_ia'] = ol_account.itemname
             audit['link'] = ol_account.itemname
-            return audit  # special case; unable to get ia acc
+
+            # This used to allow users to continue logging in with
```
Comment (bfalling) [inline openlibrary/accounts/model.py]: Let's make sure this comment describes present reality. And it'd be great if it also described what's about to happen.

File: openlibrary/templates/login.html [hunk 2]
```diff
@@ -82,7 +82,7 @@
 
 <div id="contentBody">
   $if not ctx.user:
-      <p class="xauth-switch">Open Library now uses <strong><u>email</u></strong> to login. You can use your Internet Archive or Open Library account <strong><u>email</u></strong> address.</p>
+      <p class="xauth-switch">Starting August 1st, your <a href="/account/email/forgot">Archive.org email</a> and password will be required to login to Open Library.</a></p>
```
Comment (bfalling) [inline openlibrary/templates/login.html]: Change "login" to "log in" to reflect current messaging on the site.

File: openlibrary/templates/login.html [hunk 3]
```diff
@@ -135,7 +135,7 @@
         </div>
         <div class="formElement">
             <div class="label"></div>
-            <div class="input smaller">$:_('<a href="/account/password/forgot">Forgot Password</a>?')
+            <div class="input smaller">$:_('<a href="https://archive.org/account/login.forgotpw.php">Forgot Archive.org Password</a>?')
```
Comment (bfalling) [inline openlibrary/templates/login.html]: Prior to August 1, should this "Forgot Password" link indeed be limited to returning an IA password only?

File: openlibrary/templates/login.html [hunk 4]
```diff
@@ -177,6 +177,7 @@
         "missing_fields": "Please fill out all fields and try again",
         "email_registered": "This email is already registered",
         "username_registered": "This username is already registered",
+        "ia_login_only": "Sorry, login requires using your Internet Archive email and password",
```
Comment (bfalling) [inline openlibrary/templates/login.html]: Change "login" to "log in"?

File: openlibrary/templates/login.html (hunk 3 above)
Comment (mekarpeles) [inline openlibrary/templates/login.html]: We decided to use Open Library forget password for now, as you've recommended, and on August 1st, we will only support the IA forgot password link.

This means anyone post August 1st who has not linked and does not remember their OL password will need to contact customer support.

**Comment by mekarpeles** [comment]
@bfalling fixes addressed

File: openlibrary/templates/account/password/forgot.html [hunk 5]
```diff
@@ -16,9 +22,11 @@
...
-                &nbsp;
-                <a href="javascript:;" onclick="history.go(-1);" class="large attn">$_("Cancel")</a>
             </div>
         </div>
     </form>
 </div>
+
+<div>
+
+</div>
```
Comment (bfalling) [inline openlibrary/templates/account/password/forgot.html]: What is this empty `<div>`?

Comment (mekarpeles) [inline openlibrary/templates/account/password/forgot.html]: good catch, resolved

**Pull request merged**